crates/bevy_ecs/src/event.rs

```diff
@@ -354,10 +354,37 @@ impl<'w, 's, E: Event> EventReader<'w, 's, E> {
         internal_event_reader(&mut self.last_event_count.0.clone(), &self.events).len()
     }
 
-    /// Determines if are any events available to be read without consuming any.
+    /// Determines if no events are available to be read without consuming any.
+    /// If you need to consume the iterator you can use [`EventReader::clear`].
+    ///
+    /// ## Example usage with .clear()
```

Comment:
```suggestion
    /// # Example
    /// 
    /// The following example shows a common pattern of this function in conjunction with `clear`
    /// to avoid leaking events to the next schedule iteration while also checking if it was emitted.
    /// 
```

My bad, h2 headers are weirdly rendered as h6 in the rendered docs, and it is common practice to use h1 for sections. I also added a little description for the example use case.

Reply:
I don't get what weirdness is going on on my end. Now the “Example” header is an h5 instead of h1. 😢